packages/edit-post/src/components/layout/index.native.js

```diff
@@ -128,6 +129,8 @@ class Layout extends Component {
 						style={ toolbarKeyboardAvoidingViewStyle }
 					>
 						<Header />
+						<BottomSheetSettings />
+						<BottomSheetSettings.Slot />
```

Comment:
I'm not understanding this. From what I can see, `BottomSheetSettings` is the `Fill` to `BottomSheetSettings.Slot`. If they're going to be rendered next to each other, I don't know why we'd need a slot-fill.

I think this was useful before we used the store to toggle the bottom sheet visibility, but could be simplified now?

Reply:
This pattern comes from web version. The `Sidebar` is constructed in similar way. I've picked up this solution because it do not brings a lot of additional login in `Layout` component but allows to move it over. 

Another reason is the file structure - in current implementation I import `BottomSheetSettings` with its logic from `@wordpress/block-editor` and the layout just render it.

I know that `Sidebar` is much more complex but it seems that our `BottomSheet` settings component also will be growing fast so that's the third reason behind.

However as you noticed we can took whole `BottomSheetFill` and render it in `Layout` component. After pinning some `selectors` it will also work.

You have much more experience and knowledge with this repo so let me know wdyt

Reply:
I think the web part is a bit different because there can be several sidebar fills via plugins. At least it is made clearer by it using [two different names for it](https://github.com/WordPress/gutenberg/blob/a5b1b2e271a55a7474bdd803df490871252a2d39/packages/edit-post/src/components/layout/index.js#L125):

```jsx
<SettingsSidebar />
<Sidebar.Slot />
```

So first it defines the contents of the SettingsSidebar (document + block settings) and then it renders what actually contains the sidebar. It's still a bit awkward to see them side by side but it makes sense because `SettingsSidebar` is more clearly one of possibly several things that will render in the sidebar.

I guess the question is whether we expect to support anything else other than block settings to fill that slot. With the current design, I don't think we do, because if we added a second fill, it would render alongside the block inspector (like in the sidebar), but it would present a second bottom sheet. If we wanted to support other things, we'd have to start using sidebar names (e.g. `edit-post/block`) and make sure we only [present the matching one](https://github.com/WordPress/gutenberg/blob/master/packages/edit-post/src/components/sidebar/index.js#L55) instead of just checking for `isEditorSidebarOpened`.

If we only expect to put the InspectorControls of the selected block in it, I still think we could simplify this.

Reply:
how about that ?

![image](https://user-images.githubusercontent.com/21242757/65758380-4afc6e00-e119-11e9-92f5-8fe18a952e9e.png)

Reply:
I think that looks right 👌

Reply:
ok so I will push this in a minutes